pg_search/src/postgres/customscan/pdbscan/projections/window_agg.rs

```diff
@@ -0,0 +1,527 @@
+/// a particular window function construct.
+///
+/// When a feature is fully implemented and stable, its flag should be set to `true`.
+pub mod window_functions {
+    /// Only allow window function replacement in TopN queries (with ORDER BY and LIMIT).
+    /// When true, window functions are only replaced with window_func in TopN execution context.
+    /// When false, window functions can be replaced in any query context.
+    pub const ONLY_ALLOW_TOP_N: bool = true;
+
+    /// Enable support for window functions in subqueries.
+    pub const SUBQUERY_SUPPORT: bool = false;
+
+    /// Enable support for window functions in queries with HAVING clauses.
+    pub const HAVING_SUPPORT: bool = false;
+
+    /// Enable support for window functions in queries with JOINs.
+    pub const JOIN_SUPPORT: bool = false;
+
+    /// Enable support for `FILTER` clause in window functions.
+    pub const WINDOW_AGG_FILTER_CLAUSE: bool = false;
+}
+
+/// Information about a window aggregate to compute during TopN execution
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct WindowAggregateInfo {
+    /// Target entry index where this aggregate should be projected
+    pub target_entry_index: usize,
+    /// Aggregation specification (shared with aggregatescan)
+    pub agg_spec: AggregationSpec,
+}
+
+impl WindowAggregateInfo {
+    pub fn result_type_oid(&self) -> pg_sys::Oid {
+        self.agg_spec.result_type_oid()
+    }
+
+    /// Check if we can handle this aggregation specification
+    ///
+    /// This is primarily used by window functions to check feature flag support.
+    /// Execution capability is determined by feature flags.
+    pub fn is_supported(agg_spec: &Option<AggregationSpec>) -> bool {
+        if agg_spec.is_none() {
+            return false;
+        }
+        let agg_spec = agg_spec.as_ref().unwrap();
+
+        // Check if all aggregate functions are supported
+        for agg_type in &agg_spec.agg_types {
+            // Check if this aggregate has a filter
+            let has_filter = agg_type.has_filter();
+            if has_filter && !window_functions::WINDOW_AGG_FILTER_CLAUSE {
+                return false;
+            }
+        }
+
+        // Note: PARTITION BY and ORDER BY in OVER clauses are not supported in our use case,
+        // because we compute facets over the entire result set, not partitioned subsets.
+        // If grouping_columns is non-empty, we reject the query.
+        if !agg_spec.grouping_columns.is_empty() {
+            return false;
+        }
+
+        // All required features are supported
+        true
+    }
+}
+
+/// Extract window aggregates from a query with all-or-nothing support
+///
+/// This function implements an all-or-nothing approach: either ALL window functions
+/// in the query are supported (and get replaced with window_func placeholders),
+/// or NONE of them are replaced and PostgreSQL handles all window functions with
+/// standard execution.
```

Comment:
If our explicit `paradedb.agg` function is used, then the custom scan _must_ be used, right? I didn't find code implementing this.